target/mips/translate.c

```diff
@@ -6594,12 +6593,10 @@ static void gen_compute_branch(DisasContext *ctx, uint32_t opc,
         }
         gen_load_gpr(btarget, rs);
 #ifdef TARGET_CHERI
-        /* Add PCC.base to rs */
-        tcg_gen_ld_i64(t1, cpu_env, offsetof(CPUMIPSState, active_tc.PCC) +
-                offsetof(cap_register_t, cr_base));
-        tcg_gen_add_i64(btarget, btarget, t1);
-        GEN_CCHECK_BTARGET(cpu_env);
+        /* Add PCC.base to rs (jr/jalr is relative to PCC) */
+        tcg_gen_addi_tl(btarget, btarget, ctx->pcc_base);
```

Comment:
Given this is now a constant, presumably TCG is smart enough to optimise this away and it doesn't need to be guarded by TARGET_CHERI if you use your new helper function? (Especially since you've moved the `GEN_CCHECK_BTARGET` out of the TARGET_CHERI check.)

Reply:
tcg_gen_addi_i64() is not an inline function so it requires LTO to be optimized out. I'll keep it in the ifdef for now.